File: src/TestPsychologique/models/Interpretation.java
package TestPsychologique.models;

public class Interpretation {
    private int id;
    private int testId;
    private int scoreMin;
    private int scoreMax;
    private String titre;
    private String description;
    private String conseils;

    // Constructeurs
    public Interpretation() {
    }

    public Interpretation(int testId, int scoreMin, int scoreMax, String titre, String description, String conseils) {
        this.testId = testId;
        this.scoreMin = scoreMin;
        this.scoreMax = scoreMax;
        this.titre = titre;
        this.description = description;
        this.conseils = conseils;
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getTestId() {
        return testId;
    }

    public void setTestId(int testId) {
        this.testId = testId;
    }

    public int getScoreMin() {
        return scoreMin;
    }

    public void setScoreMin(int scoreMin) {
        this.scoreMin = scoreMin;
    }

    public int getScoreMax() {
        return scoreMax;
    }

    public void setScoreMax(int scoreMax) {
        this.scoreMax = scoreMax;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getConseils() {
        return conseils;
    }

    public void setConseils(String conseils) {
        this.conseils = conseils;
    }

    @Override
    public String toString() {
        return "Interpretation{" +
                "id=" + id +
                ", testId=" + testId +
                ", scoreMin=" + scoreMin +
                ", scoreMax=" + scoreMax +
                ", titre='" + titre + '\'' +
                '}';
    }
}

File: src/Entity/Interpretation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\InterpretationRepository;

#[ORM\Entity(repositoryClass: InterpretationRepository::class)]
#[ORM\Table(name: 'interpretations')]
class Interpretation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): self
    {
        $this->id = $id;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $test_id = null;

    public function getTest_id(): ?int
    {
        return $this->test_id;
    }

    public function setTest_id(int $test_id): self
    {
        $this->test_id = $test_id;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $score_min = null;

    public function getScore_min(): ?int
    {
        return $this->score_min;
    }

    public function setScore_min(int $score_min): self
    {
        $this->score_min = $score_min;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $score_max = null;

    public function getScore_max(): ?int
    {
        return $this->score_max;
    }

    public function setScore_max(int $score_max): self
    {
        $this->score_max = $score_max;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $titre = null;

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: false)]
    private ?string $description = null;

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $conseils = null;

    public function getConseils(): ?string
    {
        return $this->conseils;
    }

    public function setConseils(?string $conseils): self
    {
        $this->conseils = $conseils;
        return $this;
    }

}
